src/openfe_analysis/reader.py

```diff
@@ -1,66 +1,99 @@
 import netCDF4 as nc
 from openff.units import unit
+import numpy as np
+import yaml
 from typing import Optional
 
 
-from . import handle_trajectories
+from openfe_analysis.utils import multistate, serialization
+from openfe_analysis.utils.multistate import _determine_position_indices
 
 
-def _determine_dt(ds) -> float:
-    # first grab integrator timestep
-    mcmc_move_data = ds.groups['mcmc_moves']['move0'][0].split('\n')
-    in_timestep = False
-    for line in mcmc_move_data:
-        if line.startswith('timestep'):
-            in_timestep = True
-        if in_timestep and line.strip().startswith('value'):
-            timestep = float(line.split()[-1]) / 1000.  # convert to ps
-            break
-    else:
-        raise ValueError("Didn't find timestep")
-    # next get the save interval
-    option_data = ds.variables['options'][0].split('\n')
-    for line in option_data:
-        if line.startswith('online_analysis_interval'):
-            nsteps = float(line.split()[-1])
-            break
-    else:
-        raise ValueError("Didn't find online_analysis_interval")
+def _determine_iteration_dt(dataset) -> float:
+    """
+    Find out the timestep between each frame in the trajectory.
+
+    Parameters
+    ----------
+    dataset : nc.Dataset
+      Dataset holding the multistatereporter generated NetCDF file.
+
+    Returns
+    -------
+    float
+      The timestep in units of picoseconds.
+
+    Raises
+    ------
+    KeyError
+      If either `timestep` or `n_steps` cannot be found in the
+      zeroth MCMC move.
+
+    Notes
+    -----
+    This assumes an MCMC move which serializes in a manner similar
+    to `openmmtools.mcmc.LangevinDynamicsMove`, i.e. it must have
+    both a `timestep` and `n_steps` defined.
+    """
+    # Deserialize the MCMC move information for the 0th entry.
+    mcmc_move_data = yaml.load(
+        dataset.groups['mcmc_moves']['move0'][0],
+        Loader=serialization.UnitedYamlLoader,
+    )
+
+    try:
+        dt = mcmc_move_data['n_steps'] * mcmc_move_data['timestep']
+    except KeyError:
+        msg = "Either `n_steps` or `timestep` are missing from the MCMC move"
+        raise KeyError(msg)
 
-    return timestep * nsteps
+    return dt.to('picosecond').m
 
 
 class FEReader(ReaderBase):
-    """A MDAnalysis Reader for nc files created by openfe RFE Protocol
+    """A MDAnalysis Reader for NetCDF files created by
+    `openmmtools.multistate.MultistateReporter`
 
-    Looks along a multistate .nc file along one of two axes:
-    - constant state/lambda (varying replica)
-    - constant replica (varying lambda)
+    Looks along a multistate NetCDF file along one of two axes:
+      - constant state/lambda (varying replica)
+      - constant replica (varying lambda)
     """
     _state_id: Optional[int]
     _replica_id: Optional[int]
     _frame_index: int
     _dataset: nc.Dataset
     _dataset_owner: bool
 
-    format = 'openfe RFE'
+    format = 'MultistateReporter'
```

Comment:
```suggestion
    format = 'MultiStateReporter'
```
The naming is not consistent, sometimes the S is capitalized, sometimes not.